app/scripts/modules/core/delivery/service/execution.service.js

```diff
@@ -115,6 +115,48 @@ module.exports = angular.module('spinnaker.core.delivery.executions.service', [
       return deferred.promise;
     }
 
+    function pauseExecution(application, executionId) {
+      var deferred = $q.defer();
+      var matcher = (execution) => {
+        return execution.status === 'PAUSED';
+      };
+
+      $http({
+        method: 'PUT',
+        url: [
+          settings.gateUrl,
+          'pipelines',
+          executionId,
+          'pause',
+        ].join('/')
+      }).then(
+        () => waitUntilExecutionMatches(executionId, matcher).then(application.reloadExecutions()).then(deferred.resolve),
```

Comment:
`application.reloadExecutions()` should be `application.reloadExecutions` - otherwise, it will fire immediately after the `PUT` returns.